enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+With this change, OLM will need to be updated to manage the `ServiceMonitor` and `PrometheusRule` objects through the lifecycle of an operator.
+
+#### ServiceMonitor and PrometheusRule Support
+
+As OLM gains support for these Prometheus objects, there is a chance that installed operators may overload the Prometheus instance. It is important that a Cluster Admin is alerted that an operator will report metrics prior to deploying the operator. As such, we propose adding a `monitoringSupported` boolean field to the `ClusterServiceVersion` object. OLM will not install either of the Prometheus object unless the `monitoringSupported` field is set to true. Additionally, Cluster Admins will need to manually approve the operator's `InstallPlans` in the following scenarios:
+
+- A fresh install for this operator is being created where this field is set to `true`
+- An existing operator had this field set to `false` and is being upgraded to a new version that sets this field to `true`
+
+> Note: The `InstallPlan` will include a helpful message for Cluster Admins if either of the situations above are encountered.
+
+On upstream clusters, there is no way for OLM to identify which Prometheus Operator ServiceAccount should be granted the RBAC privileges to view the  `ServiceMonitor` and `PrometheusRule` object created alongsde the operator. As such, the cluster admin will need to update the correct Prometheus Operator's OperatorGroup to include the correct namespace.
+
+### OpenShift Monitoring Prometheus Operator Support
+
+#### Requirements
+
+Once OLM is capable of managing `ServiceMonitor` and `PrometheusRule` objects along with the lifecycle of an operator, OLM should offer a path to automatically integrate Red Hat operators with the OpenShift Monitoring Prometheus Operator,. This integration comes with a few requirements:
+
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labeled with `openshift.io/cluster-monitoring=true` `ServiceMonitor` and `PrometheusRule` object
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the Prometheus Operator ServiceAccount with the appropriate RBAC privileges to discover the `ServiceMonitors` and `PrometheusRules` objects in the newly created namespace.
```

Comment:
why doesn't the prometheus operator's SA just have permissions to read SMs and PRs globally?

@LiliC

Reply:
probably @brancz has more historical context but this RBAC could potentially be loosened to allow access to all namespaces in a future release.